pkg/datapath/loader/tc_test.go

```diff
@@ -0,0 +1,161 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package loader
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/cilium/ebpf"
+	"github.com/cilium/ebpf/asm"
+
+	"github.com/cilium/cilium/pkg/testutils"
+	"github.com/cilium/cilium/pkg/testutils/netns"
+)
+
+func newTCProgram() (*ebpf.Program, error) {
+	return ebpf.NewProgram(&ebpf.ProgramSpec{
+		Type: ebpf.SchedCLS,
+		Instructions: asm.Instructions{
+			asm.Mov.Imm(asm.R0, 0),
+			asm.Return(),
+		},
+		License: "Apache-2.0",
+	})
+}
+
+func mustTCProgram(tb testing.TB) *ebpf.Program {
+	p, err := newTCProgram()
+	if err != nil {
+		tb.Skipf("tc programs not supported: %s", err)
+	}
+	tb.Cleanup(func() {
+		p.Close()
+	})
+	return p
+}
+
+func mustTCProgramWithName(t *testing.T, name string) *ebpf.Program {
+	p, err := ebpf.NewProgram(&ebpf.ProgramSpec{
+		Name: name,
+		Type: ebpf.SchedCLS,
+		Instructions: asm.Instructions{
+			asm.Mov.Imm(asm.R0, 0),
+			asm.Return(),
+		},
+		License: "Apache-2.0",
+	})
+	if err != nil {
+		t.Skipf("tc programs not supported: %s", err)
+	}
+	t.Cleanup(func() {
+		p.Close()
+	})
+	return p
+}
+
+func TestAttachDetachSKBProgram(t *testing.T) {
+	testutils.PrivilegedTest(t)
+
+	ns := netns.NewNetNS(t)
+	ns.Do(func() error {
+		prog := mustTCProgram(t)
+		linkDir := testutils.TempBPFFS(t)
+
+		require.NoError(t, attachSKBProgram(lo, prog, "test", linkDir, directionToParent(dirEgress), true))
+		require.NoError(t, detachSKBProgram(lo, "test", linkDir, directionToParent(dirEgress)))
+
+		return nil
+	})
+}
+
+// Upgrade a legacy tc program to tcx.
+func TestAttachSKBUpgrade(t *testing.T) {
```

Comment:
This test should be skipped if TCX is not supported, similar to the other tests in tcx_test.go.